Framework/Geometry/src/Instrument/CompAssembly.cpp

```diff
@@ -305,6 +306,9 @@ CompAssembly::getComponentByName(const std::string &cname, int nlevels) const {
     }
 
     auto rectDet = boost::dynamic_pointer_cast<const RectangularDetector>(node);
```

Comment:
Seems wrong to me that the `CompAssembly` should need to do this, but I appreciate that the capability query was already happening.